support/ebpf/dotnet_tracer.ebpf.c

```diff
@@ -107,7 +106,9 @@ dotnet_find_code_start(PerCPURecord *record, DotnetProcInfo *vi, u64 pc, u64 *co
       // convert it from u32 to u64 offset
       int pos64 = (map_elements - 2) / 2;
       u64 val64 = scratch->map64[--pos64];
-#pragma unroll 256
+#if !defined(TESTING_COREDUMP)
+  #pragma unroll 256
+#endif
```

Comment:
Avoid code repetition by defining a macro (e.g. `UNROLL`), which is empty for gcc.
Just in case, https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/613/files#r2218507693 can't be solved to use the same compiler everywhere.